src/components/NoteNameEditor/NoteNameEditorComponent.js

```diff
@@ -0,0 +1,56 @@
+import React, { Component } from "react";
+import PropTypes from "prop-types";
+import _ from "underscore";
+
+import { HeaderComponent } from 'components/NoteNameEditor/HeaderComponent/HeaderComponent';
+import {
+  StyledNoteEditorTextArea,
+  StyledNoteNameEditor
+} from "components/NoteNameEditor/NoteNameEditorStyles";
+
+export class NoteNameEditorComponent extends Component {
+  state = { name: "", propName: "" };
+
+  ref = React.createRef();
+
+  debouncedOnChange = _.debounce(name => this.props.onChange(name), 1500);
+
+  render() {
+    const { onDeleteClick, onUploadFileClick } = this.props;
+
+    return (
+      <StyledNoteNameEditor>
+        <HeaderComponent
+          onDeleteClick={onDeleteClick}
+          onUploadFileClick={onUploadFileClick}
+        />
+        <StyledNoteEditorTextArea
```

Comment:
Are we now going to have every component prefixed with "Styled"? :) Whenever we need any styles.

Reply:
Yes, It helps us to understand, that we are working with styled components, as we used with components parts like (*Component, *Container, etc.). This components are specific and sometime should receive some props to work properly with themes or styles overriding.